packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,336 @@
+    // Parse through pathsToSize to create a map object with tree-like structure.
+    for (final List<String> paths in pathsToSize.keys) {
+      for (final String path in paths) {
+        if (!currentLevel.containsKey('children')) {
+          currentLevel['children'] = <Map<String, dynamic>>[];
+        }
+
+        // TODO(peterdjlee): Optimize children look up time to O(1) by using a
+        // set instead of a list for children field.
+        Map<String, dynamic> childWithPathAsName =
+            currentLevel['children'].firstWhere(
+          (Map<String, dynamic> child) => child['n'] == path,
+          orElse: () => null,
+        ) as Map<String, dynamic>;
+
+        if (childWithPathAsName == null) {
+          childWithPathAsName = <String, dynamic>{'n': path, 'value': 0};
+
+          if (path.endsWith('libapp.so')) {
+            childWithPathAsName['n'] += ' (Dart AOT)';
+          } else if (path.endsWith('libflutter.so')) {
+            childWithPathAsName['n'] += ' (Flutter Engine)';
+          }
+
+          // TODO(peterdjlee): Include corresponding aot size data for all 3 platforms.
+          // Convert aotSizeJson to a tree-like structure and include it as children of libapp.so.
+          if (paths.contains('arm64-v8a') && path == 'libapp.so') {
+            childWithPathAsName['children'] = treemapFromJson(aotSizeJson)['children'];
+          }
+          currentLevel['children'].add(childWithPathAsName);
+        }
+        childWithPathAsName['value'] += pathsToSize[paths];
+        currentLevel = childWithPathAsName;
+      }
+      currentLevel = root;
+    }
+    return root;
+  }
+
+  // Prints all the paths from level to
+  void _printLibDetails(
+    Map<String, dynamic> currentPath,
+    String totalPath,
+    File aotSizeJson,
+  ) {
+    final String name = currentPath['n'] as String;
+    totalPath += name;
+
+    if (currentPath.containsKey('children') && !name.contains('libapp.so')) {
+      for (final Map<String, dynamic> child
+          in currentPath['children'] as List<Map<String, dynamic>>) {
+        _printLibDetails(child, totalPath + '/', aotSizeJson);
+      }
+    } else {
+      // Print total path and size if currentPath does not have any chilren.
+      _printEntitySize(totalPath, (currentPath['value'] as int) ~/ 1024, 2);
+      if (totalPath.contains('lib/arm64-v8a/libapp.so')) {
+        _analyzeAotSize(aotSizeJson);
+      }
+    }
+  }
+
+  // Go through the AOT gen snapshot size JSON and print out a collapsed summary
+  // for the first package level.
+  void _analyzeAotSize(File aotSizeJson) {
+    final SymbolNode root = _parseSymbols(
+      json.decode(aotSizeJson.readAsStringSync()) as List<dynamic>,
+    );
+
+    final int totalSymbolSize = root.children.fold(
+      0,
+      (int previousValue, SymbolNode element) => previousValue + element.value,
+    );
+
+    _printEntitySize(
+      'Dart AOT symbols accounted decompressed size',
+      totalSymbolSize ~/ 1024,
```

Comment:
add a helper for display in KB. I expect we don't want to use ~/1024 and instead want to round to the nearest KB.